src/main/java/liquibase/ext/spanner/SpannerDropAllForeignKeyConstraintsChange.java

```diff
@@ -0,0 +1,53 @@
+package liquibase.ext.spanner;
+
+import com.google.common.base.MoreObjects;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import liquibase.change.ChangeMetaData;
+import liquibase.change.DatabaseChange;
+import liquibase.change.core.DropAllForeignKeyConstraintsChange;
+import liquibase.change.core.DropForeignKeyConstraintChange;
+import liquibase.database.Database;
+import liquibase.database.jvm.JdbcConnection;
+import liquibase.exception.DatabaseException;
+import liquibase.exception.UnexpectedLiquibaseException;
+import liquibase.statement.SqlStatement;
+
+@DatabaseChange(name="dropAllForeignKeyConstraints", description = "Drops all foreign key constraints for a table", priority = ChangeMetaData.PRIORITY_DATABASE, appliesTo = "table")
+public class SpannerDropAllForeignKeyConstraintsChange extends DropAllForeignKeyConstraintsChange {
+
+  @Override
+  public boolean supports(Database database) {
+    return (database instanceof CloudSpanner);
+  }
+  
+  public SqlStatement[] generateStatements(Database database) {
+    List<SqlStatement> sqlStatements = new ArrayList<>();
+    
+    JdbcConnection connection = (JdbcConnection) database.getConnection();
+    try (PreparedStatement ps = connection.prepareStatement("SELECT CONSTRAINT_NAME FROM INFORMATION_SCHEMA.TABLE_CONSTRAINTS WHERE TABLE_CATALOG=? AND TABLE_SCHEMA=? AND TABLE_NAME=? AND CONSTRAINT_TYPE='FOREIGN KEY'")) {
```

Comment:
One of the limitations that we hit earlier on was that the INFORMATION_SCHEMA is only available in a read-only transaction, rather than read-write, and Liquibase didn't seem to care. Also that this behaviour was different between the real Spanner and the emulator. (Only the real Spanner cared)

Has this been tested against the emulator and/or real Spanner?

Reply:
I've added an integration test for it:
1. This works on real Spanner.
1. The emulator test seems to be hitting what might be a bug in the emulator: After creating a table with a foreign key, all queries on `INFORMATION_SCHEMA` will hang forever.

Reply:
Great - thanks!